### ISSUE TITLE
Add support to tool decorator for class methods

### ISSUE THREAD

**Issue by paulrobello** [jlowin/fastmcp#55]
It would be useful if I could decorate methods on a class to be used as tools.
I have lots of utility classes especially for database related function.
currently if I decorate a method "self" shows up a a parameter.

**Comment by jlowin** [comment]
Totally agree

**Comment by jlowin** [comment]
ok I think some good news is that while I am improving this, the basic pattern may not change. 

The following is likely **invalid** no matter what:

```python
from fastmcp import FastMCP

mcp = FastMCP()

class MyClass:
    @mcp.tool()
    def add(self, x, y):
        return x + y
```

The reason this is invalid is that when the decorator is evaluated, the MCP is going to load the *unbound* `MyClass.add` method no matter what. Thereafter, it won't be accessed as part of the parent class and so even a descriptor protocol can't properly bind it. Instead, a valid pattern even in today's code is to do the following:

```python
from fastmcp import FastMCP

mcp = FastMCP()

class MyClass:
    def add(self, x, y):
        return x + y

obj = MyClass()
mcp.add_tool(obj.add)

await mcp.call_tool('add', {'x': 1, 'y': 2}) # 3
```

In other words it may never make sense to decorate and use an instance method as a tool on an unbound method. 

This turns out to be true for classmethods as well -- since the function is taken *at the moment of decoration* and added to the MCP server's tool list, there isn't a way to capture the parent class either. However, the above pattern of adding a method to the MCP after it has been created still works (and could probably be added to a class as a post-initialization step)

Staticmethods work!

**Comment by jlowin** [comment]
New documentation on this topic: https://gofastmcp.com/patterns/decorating-methods
